OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -209,7 +209,26 @@ public class Aircraft : PausableConditionalTrait<AircraftInfo>, ITick, ISync, IF
 		IOverrideAircraftLanding overrideAircraftLanding;
 
 		[Sync]
-		public WAngle Facing { get; set; }
+		WAngle facing;
+		WRot orientation;
+		bool orientationDirty = true;
+
+		public WAngle Facing
+		{
+			get { return facing; }
+			set { facing = value; orientationDirty = true; }
+		}
+
+		public WRot Orientation
+		{
+			get
+			{
+				if (orientationDirty)
+					orientation = new WRot(WAngle.Zero, WAngle.Zero, Facing);
+
+				return orientation;
+			}
+		}
```

Comment:
This seems like an unnecessarily roundabout way of doing things. Why not just do something like:

```C#
		public WAngle Facing
		{
			get { return Orientation.Yaw; }
			set { Orientation = Orientation.WithYaw(value); }
		}

		public WRot Orientation { get; set; }
```

and let `Fly` set `aircraft.Orientation` directly?